qutip/tests/solver/test_brmesolve.py

```diff
@@ -0,0 +1,294 @@
+    np.testing.assert_allclose(me_num, brme_num, atol=1e-2)
+
+
+def test_jaynes_cummings_zero_temperature():
+    """
+    brmesolve: Jaynes-Cummings model, zero temperature
+    """
+    N = 10
+    a = qutip.tensor(qutip.destroy(N), qutip.qeye(2))
+    sp = qutip.tensor(qutip.qeye(N), qutip.sigmap())
+    psi0 = qutip.ket2dm(qutip.tensor(qutip.basis(N, 1), qutip.basis(2, 0)))
+    a_ops = [(a + a.dag(), qutip.coefficient(lambda t, w: kappa * (w >= 0)))]
+    e_ops = [a.dag()*a, sp.dag()*sp]
+
+    w0 = 1.0 * 2*np.pi
+    g = 0.05 * 2*np.pi
+    kappa = 0.05
+    times = np.linspace(0, 2 * 2*np.pi / g, 1000)
+
+    c_ops = [np.sqrt(kappa) * a]
+    H = w0*a.dag()*a + w0*sp.dag()*sp + g*(a+a.dag())*(sp+sp.dag())
+
+    me = qutip.mesolve(H, psi0, times, c_ops, e_ops)
+    brme = brmesolve(H, psi0, times, a_ops, e_ops)
+    for me_expectation, brme_expectation in zip(me.expect, brme.expect):
+        # Accept 5% error.
+        np.testing.assert_allclose(me_expectation, brme_expectation, atol=5e-2)
+
+
+def test_tensor_system():
+    """
+    brmesolve: Check for #572 bug.
+    """
+    w1, w2, w3 = 1, 2, 3
+    gamma2, gamma3 = 0.1, 0.1
+    id2 = qutip.qeye(2)
+
+    # Hamiltonian for three uncoupled qubits
+    H = (w1/2. * qutip.tensor(qutip.sigmaz(), id2, id2)
+         + w2/2. * qutip.tensor(id2, qutip.sigmaz(), id2)
+         + w3/2. * qutip.tensor(id2, id2, qutip.sigmaz()))
+
+    # White noise
+    S2 = qutip.coefficient(lambda t, w: gamma2, args={'w':0})
+    S3 = qutip.coefficient(lambda t, w: gamma3, args={'w':0})
+
+    qubit_2_x = qutip.tensor(id2, qutip.sigmax(), id2)
+    qubit_3_x = qutip.tensor(id2, id2, qutip.sigmax())
+
+    # Initial state : first qubit is excited
+    grnd2 = qutip.sigmam() * qutip.sigmap()  # 2x2 ground
+    exc2 = qutip.sigmap() * qutip.sigmam()   # 2x2 excited state
+    ini = qutip.tensor(exc2, grnd2, grnd2)   # Full system
+
+    # Projector on the excited state of qubit 1
+    proj_up1 = qutip.tensor(exc2, id2, id2)
+    times = np.linspace(0, 10./gamma3, 1000)
+
+    sol = brmesolve(H, ini, times, [[qubit_2_x, S2], [qubit_3_x, S3]],
+                    e_ops=[proj_up1]).expect
+
+    np.testing.assert_allclose(sol[0], np.ones_like(times))
+
+
+def test_solver_accepts_list_hamiltonian():
+    """
+    brmesolve: input list of Qobj
+    """
+    delta = 0.0 * 2 * np.pi
+    epsilon = 0.5 * 2 * np.pi
+    gamma = 0.25
+    c_ops = [np.sqrt(gamma) * qutip.sigmam()]
+    e_ops = pauli_spin_operators()
+    H = [delta * 0.5 * qutip.sigmax(), epsilon * 0.5 * qutip.sigmaz()]
+    psi0 = (2 * qutip.basis(2, 0) + qutip.basis(2, 1)).unit()
+    times = np.linspace(0, 10, 100)
+    me = qutip.mesolve(H, psi0, times, c_ops=c_ops, e_ops=e_ops).expect
+    brme = brmesolve(H, psi0, times, [], e_ops=e_ops, c_ops=c_ops).expect
+    for me_expectation, brme_expectation in zip(me, brme):
+        np.testing.assert_allclose(me_expectation, brme_expectation, atol=1e-8)
+
+
+# test_time_dependence
+
```

Comment:
```suggestion
```
This comment seems a bit lost -- if it should stay, let's move it to the test it refers (to even better, just extend the name of the test to make it clear what the test is for).